Dependencies, Lint, RxJS Imports

kylecordes opened the pull request:
Update dependency ranges, this works with Angular 5.

Import RxJS idiomatically.

Fix various lint issues.

Commit:
feat: Update dependencies

Update dependencies on Angular and PrimeNG to avoid deprecation warnings and be compatible wit on Angular and PrimeNG to avoid deprecation warnings and be compatible wit

Commit:
refactor: Import RxJS idiomatically to minimize size

Import RxJS idiomatically, follow normal Angular practice. This avoid relying on all of RxJS,
instead relies on just the handful of needed operators.

StefH commented on demo/demo.component.ts:
```diff
@@ -1,7 +1,7 @@
 @Component({
-  selector: 'ao-demo-app',
-  template: '<ao-employee-grid-odata></ao-employee-grid-odata>'
+  selector: 'wnp-demo-app',
```
Why would you change this to `wnp` instead of `ao` (angular odata) ?

kylecordes replied:
Your lint rules require it to be wnp... I figured the lint rules were right. Should I remove/disable the lint rule instead?

StefH replied:
Ah, I see, I copy-pasted from another project, my mistake. Please change tslint to `oa`.

kylecordes replied:
OK, I just did that. A couple more changes coming shortly.

Commit:
fix(selectors): Adjust tslint rules to match 'ao' selector

Commit:
fix(package): Correct peer vs devDependencies

Project depends on RxJS as a peerDependency, just as it does Angular itself.

Also, there is no such package as angular/common/http, that is just part of the common package.

Commit:
fix: Avoid circular dependencies via barrel

rearrange imports so that there are no circular dependencies via the TypeScript barrel. Also,
remove use of "string" package, instead use regex to strip '/' characters.

kylecordes commented:
Pardon the slightly broken way I split the last couple of commits. I inadvertently removed the string dependency one commit prior to removing the need for it.  That will become moot if you decide to accept the PR and squash.

StefH commented:
Is it now ok?

kylecordes commented:
oh yes, I was just saying if you look at my two most recent commits:

https://github.com/kylecordes/angular-odata-es5/commit/43d9395dd9a6fb0117340ddd040e2ec26dde43a3

https://github.com/kylecordes/angular-odata-es5/commit/2ac452b9f184a8e71f33a25f1e5f3c9b84ae38df

I removed the "string" dependency from the package file in the first one, then I removed the need for this dependency and the second one. Certainly you should be able to do all of your normal testing and used to make sure it's all solid.
